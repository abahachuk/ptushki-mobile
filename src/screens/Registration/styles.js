import { StyleSheet } from "react-native";

/* eslint  import/prefer-default-export: 0 */
export const styles = StyleSheet.create({
  scrollContainer: {
    paddingTop: 20,
    paddingBottom: 12,
    width: "100%"
  },
  container: {
    justifyContent: "space-between",
    paddingRight: 18,
    paddingLeft: 18
  },
  headerText: {
    fontSize: 34,
    paddingBottom: 20,
    color: "#4f6e7c"
  },
  hintText: {
    marginTop: 15,
    marginBottom: 15,
    color: "#546E7A",
    fontFamily: "Roboto",
    fontSize: 14,
    lineHeight: 18,
    letterSpacing: 0.1
  },
  belowInput: {
    marginTop: 15
  },
  lastInput: {
    marginBottom: 20
  },
  footerBtn: {
    marginTop: 15
  }
});
