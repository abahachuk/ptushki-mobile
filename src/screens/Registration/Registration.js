import React, { useState } from "react";
import PropTypes from "prop-types";
import { Text, ScrollView, KeyboardAvoidingView } from "react-native";

import {
  makeValidatorEmail,
  makeValidatorPassword,
  makeRequiredValidator
} from "../../utils/validators";
import { styles } from "./styles";
import { Button, Input } from "../../components";
import { translate } from "../../i18n";
import { AuthService } from "../../api";

const fields = {
  email: "email",
  password: "password",
  firstName: "firstName",
  lastName: "lastName",
  phone: "phone"
};

const Registration = props => {
  const {
    emailDefault,
    passwordDefault,
    firstNameDefault,
    lastNameDefault,
    phoneDefault
  } = props;
  const [
    { email, password, firstName, lastName, phone },
    setRegistrationData
  ] = useState({
    [fields.email]: emailDefault,
    [fields.password]: passwordDefault,
    [fields.firstName]: firstNameDefault,
    [fields.lastName]: lastNameDefault,
    [fields.phone]: phoneDefault
  });
  const [emailError, setEmailError] = useState("");
  const [passwordError, setPasswordError] = useState("");
  const [firstNameError, setFirstNameError] = useState("");
  const [lastNameError, setLastNameError] = useState("");
  const [phoneError, setPhoneError] = useState("");

  const validateEmail = makeValidatorEmail(translate("validationError.email"));
  const validatePassword = makeValidatorPassword(
    translate("validationError.password")
  );
  const validateFirstName = makeRequiredValidator(
    translate("validationError.firstName")
  );
  const validateLastName = makeRequiredValidator(
    translate("validationError.lastName")
  );
  const validatePhone = makeRequiredValidator(
    translate("validationError.phone")
  );

  const setRegistrationDataCommon = field => value =>
    setRegistrationData(prevData => ({ ...prevData, [field]: value }));

  const onRegisterPress = () => {
    AuthService.registrate(email, password, firstName, lastName, phone)
      .then(data => {
        if (data) {
          props.navigation.navigate("registrationSuccess", {
            origin: "registrationSuccess",
            registrationSuccess: translate("registration.registrationSuccess"),
            hintText: translate("registration.hintText")
          });
        }
      })
      .catch(err => {
        // TODO: show message for user
        console.info(err.message);
      });
  };
  const onBackPress = () => {
    props.navigation.goBack();
  };
  const onAuthFieldBlur = () => {
    setEmailError(validateEmail(email));
    setPasswordError(validatePassword(password));
  };
  const onFirstNameBlur = () => {
    setFirstNameError(validateFirstName(firstName));
  };
  const onLastNameBlur = () => {
    setLastNameError(validateLastName(lastName));
  };
  const onPhoneBlur = () => {
    setPhoneError(validatePhone(phone));
  };

  return (
    <ScrollView
      contentContainerStyle={styles.scrollContainer}
      showsVerticalScrollIndicator={false}
      enabled
    >
      <KeyboardAvoidingView style={styles.container} enabled>
        <Text style={styles.headerText}>
          {translate("registration.sign-up")}
        </Text>
        <Input
          value={email}
          label={translate("registration.email")}
          textContentType="emailAddress"
          onChangeText={setRegistrationDataCommon(fields.email)}
          error={emailError}
          onTextInputBlur={onAuthFieldBlur}
        />
        <Input
          value={password}
          label={translate("registration.password")}
          textContentType="password"
          onChangeText={setRegistrationDataCommon(fields.password)}
          error={passwordError}
          onTextInputBlur={onAuthFieldBlur}
          wrapperStyles={[styles.belowInput]}
        />
        <Text style={styles.hintText}>
          {translate("registration.communicationDataHint")}
        </Text>
        <Input
          value={firstName}
          label={translate("registration.firstName")}
          textContentType="name"
          onChangeText={setRegistrationDataCommon(fields.firstName)}
          error={firstNameError}
          onTextInputBlur={onFirstNameBlur}
        />
        <Input
          value={lastName}
          label={translate("registration.lastName")}
          textContentType="familyName"
          onChangeText={setRegistrationDataCommon(fields.lastName)}
          error={lastNameError}
          onTextInputBlur={onLastNameBlur}
          wrapperStyles={[styles.belowInput]}
        />
        <Input
          value={phone}
          label={translate("registration.phone")}
          textContentType="telephoneNumber"
          onChangeText={setRegistrationDataCommon(fields.phone)}
          error={phoneError}
          onTextInputBlur={onPhoneBlur}
          wrapperStyles={[styles.belowInput, styles.lastInput]}
        />
        <Button
          caption={translate("registration.sign-up")}
          onPress={onRegisterPress}
          appearance="Dark"
        />
        <Button
          caption={translate("registration.back").toUpperCase()}
          onPress={onBackPress}
          appearance="Borderless"
        />
      </KeyboardAvoidingView>
    </ScrollView>
  );
};
Registration.navigationOptions = {
  title: "Регистрация",
  header: null
};

Registration.propTypes = {
  emailDefault: PropTypes.string,
  passwordDefault: PropTypes.string,
  firstNameDefault: PropTypes.string,
  lastNameDefault: PropTypes.string,
  phoneDefault: PropTypes.string,
  navigation: PropTypes.shape({
    navigate: PropTypes.func,
    goBack: PropTypes.func
  }).isRequired
};
Registration.defaultProps = {
  emailDefault: "",
  passwordDefault: "",
  firstNameDefault: "",
  lastNameDefault: "",
  phoneDefault: ""
};

export default Registration;
