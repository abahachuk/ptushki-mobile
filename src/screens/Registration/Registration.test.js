import React from "react";
import renderer from "react-test-renderer";

import Registration from "./Registration";
/* eslint  no-undef: 0 */
describe("Registration screen", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<Registration />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
