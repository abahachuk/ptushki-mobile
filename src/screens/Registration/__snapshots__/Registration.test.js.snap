// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration screen renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "paddingBottom": 12,
      "paddingTop": 20,
      "width": "90%",
    }
  }
  enabled={true}
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#4f6e7c",
            "fontSize": 34,
            "paddingBottom": 20,
          }
        }
      >
        [missing "en.registration.sign-up" translation]
      </Text>
      <View
        style={Array []}
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#151822",
                "borderRadius": 6,
                "borderWidth": 2,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingTop": 12,
                "position": "relative",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#090a10",
                  "fontSize": 23,
                  "left": 19,
                  "position": "absolute",
                  "top": 21,
                },
              ]
            }
          >
            [missing "en.registration.email" translation]
          </Text>
          <TextInput
            allowFontScaling={true}
            blurOnSubmit={true}
            maxLength={100}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onTextInputBlur={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#090a10",
                "fontSize": 23,
              }
            }
            textContentType="emailAddress"
            underlineColorAndroid="transparent"
            value=""
            wrapperStyles={Array []}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#b00020",
              "fontSize": 18,
            }
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 15,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#151822",
                "borderRadius": 6,
                "borderWidth": 2,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingTop": 12,
                "position": "relative",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#090a10",
                  "fontSize": 23,
                  "left": 19,
                  "position": "absolute",
                  "top": 21,
                },
              ]
            }
          >
            [missing "en.registration.password" translation]
          </Text>
          <TextInput
            allowFontScaling={true}
            blurOnSubmit={true}
            maxLength={100}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onTextInputBlur={[Function]}
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "color": "#090a10",
                "fontSize": 23,
              }
            }
            textContentType="password"
            underlineColorAndroid="transparent"
            value=""
            wrapperStyles={
              Array [
                Object {
                  "marginTop": 15,
                },
              ]
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "position": "absolute",
                  "right": 15,
                  "top": 20,
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/ic_eye_show/ic_eye_show2x.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 35,
                    "width": 35,
                  },
                ]
              }
            />
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#b00020",
              "fontSize": 18,
            }
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#546E7A",
            "fontFamily": "Roboto",
            "fontSize": 14,
            "lineHeight": 24,
            "marginBottom": 16,
            "marginLeft": 6,
          }
        }
      >
        [missing "en.registration.communicationDataHint" translation]
      </Text>
      <View
        style={Array []}
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#151822",
                "borderRadius": 6,
                "borderWidth": 2,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingTop": 12,
                "position": "relative",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#090a10",
                  "fontSize": 23,
                  "left": 19,
                  "position": "absolute",
                  "top": 21,
                },
              ]
            }
          >
            [missing "en.registration.firstName" translation]
          </Text>
          <TextInput
            allowFontScaling={true}
            blurOnSubmit={true}
            maxLength={100}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onTextInputBlur={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#090a10",
                "fontSize": 23,
              }
            }
            textContentType="name"
            underlineColorAndroid="transparent"
            value=""
            wrapperStyles={Array []}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#b00020",
              "fontSize": 18,
            }
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 15,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#151822",
                "borderRadius": 6,
                "borderWidth": 2,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingTop": 12,
                "position": "relative",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#090a10",
                  "fontSize": 23,
                  "left": 19,
                  "position": "absolute",
                  "top": 21,
                },
              ]
            }
          >
            [missing "en.registration.lastName" translation]
          </Text>
          <TextInput
            allowFontScaling={true}
            blurOnSubmit={true}
            maxLength={100}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onTextInputBlur={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#090a10",
                "fontSize": 23,
              }
            }
            textContentType="familyName"
            underlineColorAndroid="transparent"
            value=""
            wrapperStyles={
              Array [
                Object {
                  "marginTop": 15,
                },
              ]
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#b00020",
              "fontSize": 18,
            }
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginTop": 15,
            },
            Object {
              "marginBottom": 15,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#151822",
                "borderRadius": 6,
                "borderWidth": 2,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingTop": 12,
                "position": "relative",
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#090a10",
                  "fontSize": 23,
                  "left": 19,
                  "position": "absolute",
                  "top": 21,
                },
              ]
            }
          >
            [missing "en.registration.phone" translation]
          </Text>
          <TextInput
            allowFontScaling={true}
            blurOnSubmit={true}
            maxLength={100}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onTextInputBlur={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#090a10",
                "fontSize": 23,
              }
            }
            textContentType="telephoneNumber"
            underlineColorAndroid="transparent"
            value=""
            wrapperStyles={
              Array [
                Object {
                  "marginTop": 15,
                },
                Object {
                  "marginBottom": 15,
                },
              ]
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#b00020",
              "fontSize": 18,
            }
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 6,
              "paddingBottom": 13,
              "paddingTop": 13,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#546e7a",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#546e7a",
                "fontSize": 18,
                "fontWeight": "500",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          [missing "en.registration.sign-up" translation]
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 6,
              "paddingBottom": 13,
              "paddingTop": 13,
              "width": "100%",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#546e7a",
                "fontSize": 18,
                "fontWeight": "500",
                "textAlign": "center",
                "textTransform": "uppercase",
              },
              undefined,
            ]
          }
        >
          [MISSING "EN.REGISTRATION.BACK" TRANSLATION]
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
